bulkExport/uploadGlueScriptsToS3.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs';
+import * as AWS from 'aws-sdk';
+import axios from 'axios';
+
+const sendCfnResponse = async (event, status: 'SUCCESS' | 'FAILED', error?: Error) => {
+    const responseBody = JSON.stringify({
+        Status: status,
+        Reason: error?.message,
+        // The value of PhysicalResourceId doesn't really matter in this case.
+        // It just needs to be the same string on all responses to indicate that it is the same resource.
+        PhysicalResourceId: 'glueScripts',
+        StackId: event.StackId,
+        RequestId: event.RequestId,
+        LogicalResourceId: event.LogicalResourceId,
+    });
+    console.log(`Sending response to CFN: ${responseBody}`);
+    await axios.put(event.ResponseURL, responseBody);
+};
+/**
+ * Custom resource lambda handler that uploads a specific file to s3.
+ * Custom resource spec: See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-custom-resources.html
+ * @param event Custom resource request event. See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref-requests.html
+ */
+exports.handler = async event => {
+    console.log(event);
+    try {
+        if (process.env.GLUE_SCRIPTS_BUCKET === undefined) {
+            throw new Error('Missing env variable GLUE_SCRIPTS_BUCKET');
```

Comment:
If this error throws - will the deployment fail?

Could be race condition that the glue script was never uploaded to S3?

Reply:
yes, deployment will fail. that's what we want, there's no way to recover. The CFN template is wrong in that case.

No. This custom resource only succeeds after the upload to s3 is complete. CFN will wait until it gets a `SUCCESS` response from the custom resource before completing the CFN deployment.